tests/pipelines/test_pipelines_text_classification.py

```diff
@@ -94,6 +94,70 @@ def test_small_model_pt(self):
             ],
         )
 
+    ###################
+    @require_torch
+    def test_inputs_args_kwargs(self):
+        classifier = pipeline(
+            task="text-classification",
+            model="hf-internal-testing/tiny-random-distilbert",
+            framework="pt"
+        )
+
+        outputs0 = classifier(inputs=["test_args_kwargs input string",
+                                      "test_args_kwargs input string"])
+        outputs1 = classifier(["test_args_kwargs input string",
+                               "test_args_kwargs input string"])
+        self.assertTrue(outputs0 == outputs1)
+        # Fixed in this PR.
+        # Got and expected:
+        # output0 -> [{'label': 'LABEL_0', 'score': 0.5035483837127686}, {'label': 'LABEL_0', 'score': 0.5035483837127686}]
+        # output1 -> [{'label': 'LABEL_0', 'score': 0.5035483837127686}, {'label': 'LABEL_0', 'score': 0.5035483837127686}]
+
+    @require_torch
+    def test_top_k_args_kwargs(self):
+        classifier = pipeline(
+            task="text-classification",
+            model="hf-internal-testing/tiny-random-distilbert",
+            framework="pt"
+        )
+
+        # NOTE: OK.
+        outputs0 = classifier("test_args_kwargs input string")
+        outputs1 = classifier(["test_args_kwargs input string"])
+        self.assertTrue(isinstance(outputs0, dict) and isinstance(outputs1, list))
+        self.assertTrue(outputs0 == outputs1[0])
+        # Got and expected:
+        # output0 -> {'label': 'LABEL_0', 'score': 0.5035483837127686}
+        # output1 -> [{'label': 'LABEL_0', 'score': 0.5035483837127686}]
+
+        # NOTE: Not OK.
+        outputs0 = classifier("test_args_kwargs input string", top_k=2)
+        outputs1 = classifier("test_args_kwargs input string", 2)
```

Comment:
This should indeed print the `Ignoring : (2,)`  for legacy purpose we can error out I guess.